crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -2004,13 +1999,9 @@ impl<'a> Checker<'a> {
             for (expr, snapshot) in type_definitions {
                 self.semantic.restore(snapshot);
 
-                // Type definitions should only be considered "`__future__` type definitions"
-                // if they are annotations in a module where `from __future__ import
-                // annotations` is active, or they are type definitions in a stub file.
-                debug_assert!(
-                    self.semantic.future_annotations_or_stub()
```

Comment:
Curious if you've looked at other uses of `future_annotations_or_stub` -- is there any use of it that we _wouldn't_ want to apply also in a type checking block?